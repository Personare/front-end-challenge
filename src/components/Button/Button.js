import React from 'react'
import styled, { keyframes } from 'styled-components'
import {
  oneOfType, arrayOf, node, func, string
} from 'prop-types'

const Button = ({ children, onClick, type }) => (
  <CustomButton onClick={onClick} type={type}>
    {children}
  </CustomButton>
)

const zoomInOut = keyframes`
  from {
    transform: scale(0.8)
  }

  to {
    transform: scale(1)
  }
`

const CustomButton = styled.button`
    height: 100px;
    width: 100%;
    background-color: #3498DB;
    max-width: 100px;
    border-radius: 100%;
    border: none;
    position: fixed;
    bottom: 0;
    right: 0;
    margin: 20px;
    color:  #ffffff;
    animation: ${zoomInOut} 1s alternate infinite;
`

Button.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
  onClick: func.isRequired,
  type: string
}


Button.defaultProps = {
  type: 'button'
}
export default Button
