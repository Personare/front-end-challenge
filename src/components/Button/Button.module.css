.button {
  background-color: #692446;
  border-bottom: 3px solid #331a2a;
  border-left: 0;
  border-radius: 3px;
  border-right: 0;
  border-top: 0;
  color: #fff;
  cursor: pointer;
  font-size: 18px;
  font-weight: bold;
  margin-top: 30px;
  outline: none;
  padding: 20px 40px;
  text-transform: uppercase;
}

.button:active,
.button:disabled {
  border: 0;
  margin-top: 33px;
}

.button:disabled {
  cursor: initial;
  color: #333;
  opacity: 0.5;
}
