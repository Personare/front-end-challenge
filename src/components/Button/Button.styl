.Btn
  transition: all .2s ease-in-out
  display: inline-block
  height: 2.25rem
  padding: 0 1.625rem
  margin: .375rem 0
  color: #fff
  background-color: transparent
  border: none
  border-radius: 2px
  text-transform: uppercase
  font-weight: 500
  cursor: pointer
  &-default
    background-color: #ff4081
    &:hover
      box-shadow: 0 0 4px rgba(0, 0, 0, .12), 1px 3px 4px rgba(0 ,0, 0, .2)
      background-color: lighten(#ff4081, 10%)
