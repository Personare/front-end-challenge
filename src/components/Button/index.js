import React, { useContext } from "react";
import { useLocation, useNavigate } from "react-router-dom";
import GameContext from "../../context/GameContext";
import ButtonContainer from "./styles";

export default function Button() {
  const { gameOn, setGameOn } = useContext(GameContext);
  const location = useLocation();
  let navigate = useNavigate();

  function handleClick(e) {
    e.preventDefault();
    if(e.target.textContent === "Jogar") {
      setGameOn(true);
    } else {
      setGameOn(false);
      navigate("/");
    }
  }

  return (
    <ButtonContainer>
      <div onClick={e => handleClick(e)}>
        { gameOn
          ? <p>Voltar</p>
          : ( location.pathname === "/"
            ? <p>Jogar</p>
            : <></>
          )
        }
      </div>
    </ButtonContainer>
  );
}
