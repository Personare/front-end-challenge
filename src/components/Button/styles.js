import styled from "styled-components";

const ButtonContainer = styled.button`
  background: #2F4A94;
  border-radius: 8px;
  box-shadow: 0 10px 10px 0 rgba(0, 0, 0, 0.2);
  font-size: 1.5rem;
  color: #FFF;

  div {
    width: 100%;
    height: 100%;
    padding: 10px 25px;
    cursor: pointer;
  }
`;

export default ButtonContainer;
