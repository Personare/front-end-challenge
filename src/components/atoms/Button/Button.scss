.Button {
    border: none;
    border-radius: 2px;
    color: #fff;
    border: 2px solid #fff;
    background: none;
    cursor: pointer;
    display: inline-block;
    font-family: 'Roboto',arial,sans-serif;
    font-size: 16px;
    font-weight: 400;
    letter-spacing: .5px;
    line-height: 20px;
    min-height: 48px;
    min-width: 100px;
    padding: 15px 24px;
    text-align: center;
    text-decoration: none;
    text-transform: uppercase;
    will-change: background-color;
    position: relative;
    overflow: hidden;
    transition: color .5s cubic-bezier(0.19, 1, 0.22, 1);
    z-index: 1;

    &::before {
        content: '';
        display: block;
        width: 120%;
        height: 100%;
        background-color: #fff;
        position: absolute;
        top: 0;
        left: 0;
        transform-origin: 0% 100%;
        transform: rotate3d(0, 0, 1, -50deg) translate3d(0, -3em, 0);
        transition: all .4s cubic-bezier(0.19, 1, 0.22, 1);
        z-index: -1;
    }

    &:hover {
        color: #111;

        &::before {
            transform: rotate3d(0, 0, 0, 0);
        }
    }

    &.--disabled {
        border: 0;
        background-color: #b4b4b4;
        color: #757575;
        cursor: not-allowed;
        outline: 0;

        &::before {
            content: none;
        }

        &:hover {
            color: #757575;
        }
    }
}
