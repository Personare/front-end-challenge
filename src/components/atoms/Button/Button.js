import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

const Button = ({ onClick, children }) => (
    <button onClick={onClick} className="Button">
        {children}
    </button>
);

Button.propTypes = {
    onClick: PropTypes.func,
    children: PropTypes.string
};

export default Button;
