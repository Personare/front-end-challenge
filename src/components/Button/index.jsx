import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.module.css'

const Button = ({
  isLoading,
  setIsShuffle,
  setShuffleCards,
  cards,
  isShuffle,
  text,
}) => {
  const handleClick = () => {
    setIsShuffle(true)
    setShuffleCards(cards)
  }

  return (
    <button
      className={styles.button}
      onClick={handleClick}
      disabled={isLoading || isShuffle || !cards.length}
    >
      {text}
    </button>
  )
}
Button.defaultProps = {
  text: 'Embaralhar cartas',
}

Button.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  text: PropTypes.string,
  setIsShuffle: PropTypes.func.isRequired,
  setShuffleCards: PropTypes.func.isRequired,
  isShuffle: PropTypes.bool.isRequired,
  cards: PropTypes.array.isRequired,
}

export default Button
