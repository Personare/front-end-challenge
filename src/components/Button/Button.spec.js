import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import sinon from 'sinon'
import Button from './Button'

describe('Button', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Button onClick={Function} type="button">
        Button text
      </Button>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with default props', () => {
    const tree = renderer.create(
      <Button onClick={Function}>
        Button text
      </Button>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should call onClick function', () => {
    const spy = sinon.spy()
    const tree = mount(
      <Button onClick={spy}>
        Button text
      </Button>
    )

    tree.find('button').simulate('click')

    expect(spy.called).toMatchSnapshot()
  })
})
