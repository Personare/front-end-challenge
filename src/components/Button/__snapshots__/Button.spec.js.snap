// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should call onClick function 1`] = `true`;

exports[`Button should render correctly 1`] = `
<button
  className="sc-bdVaJa duNfdq"
  onClick={[Function]}
  type="button"
>
  Button text
</button>
`;

exports[`Button should render correctly with default props 1`] = `
<button
  className="sc-bdVaJa duNfdq"
  onClick={[Function]}
  type="button"
>
  Button text
</button>
`;
